pkg/controller/gc/gc_controller_test.go

```diff
@@ -62,33 +62,37 @@ func (f *FakePodControl) clear() {
 func TestGC(t *testing.T) {
 
 	testCases := []struct {
-		pods            map[string]api.PodPhase
-		threshold       int
-		deletedPodNames sets.String
+		pods                 map[string]api.PodPhase
+		threshold            int
+		numDeletedPods       int
+		validDeletedPodNames sets.String
 	}{
 		{
 			pods: map[string]api.PodPhase{
 				"a": api.PodFailed,
 				"b": api.PodSucceeded,
 			},
-			threshold:       0,
-			deletedPodNames: sets.NewString("a", "b"),
+			threshold:            0,
+			numDeletedPods:       2,
+			validDeletedPodNames: sets.NewString("a", "b"),
 		},
 		{
 			pods: map[string]api.PodPhase{
 				"a": api.PodFailed,
 				"b": api.PodSucceeded,
 			},
-			threshold:       1,
-			deletedPodNames: sets.NewString("a"),
+			threshold:            1,
+			numDeletedPods:       1,
+			validDeletedPodNames: sets.NewString("a", "b"),
```

Comment:
I don't think this is correct. "a" should always be the pod that is deleted since it's creationTimestamp is before "b"'s (see line 107). If that's not happening than this is a bug.